readinglist/resource.py

```diff
@@ -270,11 +282,110 @@ def _extract_sorting(self, queryparams):
                     }
                     self.raise_invalid(**error_details)
 
+                if field == 'last_modified':
+                    last_modified_used = True
+
                 direction = -1 if order == '-' else 1
                 sorting.append((field, direction))
 
+        if not last_modified_used:
+            sorting.append(('last_modified', -1))
         return sorting
 
+    def _extract_pagination_token(self, queryparams):
+        """Get pagination params."""
+        settings = self.request.registry.settings
+        try:
+            limit = int(queryparams.get('_limit', settings.get(
+                'readinglist.paginate_by', DEFAULT_PAGINATE_BY)))
+        except ValueError:
+            error_details = {
+                'name': None,
+                'location': 'querystring',
+                'description': "_limit should be an integer"
+            }
+            self.request.errors.add(**error_details)
+            raise errors.json_error(self.request.errors)
+
+        token = queryparams.get('_token', None)
+        filters = []
+        if token:
+            rules = json.loads(b64decode(token))
+            for rule in rules:
+                filters.append(self._extract_filters(rule))
+        return filters, limit
+
+    def _next_page_url(self, sorting, limit, last_record):
+        queryparams = self.request.GET.copy()
+        queryparams['_limit'] = limit
+        queryparams['_token'] = self._build_pagination_token(
+            sorting, last_record)
+        return '%s%s?%s' % (self.request.host_url, self.request.path_info,
+                            urlencode(queryparams))
+
+    def _build_pagination_token(self, sorting, last_record):
+        """Build a continuation filtering rules.
+
+        Something like that when sorting::
+
+            (status == 2 and last_modified < 123) OR status < 2
+
+
+        Something like that when sorting only per last_modified::
+
+            last_modified < 123
+
+        """
+        first_sort_field = None
+        last_sort_field = None
+        if len(sorting) > 1:
+            first_sort_field, first_sort_direction = sorting[0]
+            last_sort_field, last_sort_direction = sorting[-1]
+            if last_sort_field == 'last_modified':
+                last_sort_field, last_sort_direction = sorting[-2]
+
+        try:
+            _, last_modified_direction = [x for x in sorting
+                                          if x[0] == 'last_modified'][0]
+        except IndexError:
+            last_modified_direction = 1
```

Comment:
Instead of assigning ``-1`` and ``1`` and compare to ``1`` and ``-1`` in the code, **maybe** it would be readable to rename to ``last_modified_desc`` and assign with True or False.